pkg/eth/tx_listener.go

```diff
@@ -88,11 +95,23 @@ func (t *TxListener) PollForTxCompletion(txID string, updates chan<- string) {
 
 }
 
-func (t *TxListener) checkTx(hash common.Hash) (bool, error) {
+func (t *TxListener) waitForTimeout() {
+	c := time.After(t.timeout)
```

Comment:
Minor and doesn't really need to be changed, but I think this could also be `<- time.After(t.timeout)`.  Or you can use a `select` too (but prob too verbose for this purpose).

Reply:
I also wonder if this needs a quit channel or something to kill it upon success/fail of the transaction.  Otherwise there might be a whole bunch of goroutines still waiting on the timer.